lib/matplotlib/animation.py

```diff
@@ -246,7 +246,7 @@ def frame_size(self):
         width_inches, height_inches = self.fig.get_size_inches()
         return width_inches * self.dpi, height_inches * self.dpi
 
-    def setup(self, fig, outfile, dpi, *args):
+    def setup(self, fig, outfile, dpi, *args, **kwargs):
```

Comment:
It was added way back in 2012 in the original work-in-progress [commit](/matplotlib/matplotlib/commit/f4ed2a229764ac63eba6e7bbf2d08235ee1b1b58) for animation file writers. Unfortunately, that commit has no further documentation other than the code changes themselves. I would venture a guess though that the reasoning was that `saving()` took arbitrary arguments and called `setup()` with them so, if for some reason an overriden `setup()` called the super `setup()` with the entire set of arbitrary arguments, it wouldn't break. If that is true then I think we should remove it as that case is unlikely and would be kind of dirty anyway.